### ISSUE TITLE
Releasing MIDI key is treated as second press

### ISSUE THREAD

**Issue by fyrkantis** [k5md/M2KB#7]
The incoming `note off` event is enterpreted the same as `note on`. This causes issues with other applications thinking the key is still pressed until I reset it on my normal keyboard, and also playing the note a second time when I release. The `note off` should preferably trigger a `key up`, or nothing at all.

(lovely tool btw, I'm using it for sending input to ComputerCraft)

**Comment by k5md** [comment]
Hi, @fyrkantis ! I haven't touched any midi-related stuff in a while, so huge thanks for the fix! I'll test @ merge it tomorrow (and fix the building pipeline as well)

**Comment by fyrkantis** [comment]
No problem @k5md! I am very grateful that you made this in the first place. It is exactly what I need!

**Comment by k5md** [comment]
Hello again! I've checked the patch. While the original version works as expected, running the patched one on my mkb results in sending mapped keyboard key both on press (no matter what velocity is) and release. That's the same result that you've got with original version, I guess.
![patchm2kb](https://github.com/user-attachments/assets/6739d4a0-422d-4d1b-8d14-d189f732be3f)
The mkb in question generates correct midi-off event on key release:
![midiox](https://github.com/user-attachments/assets/50757862-ee7d-45fd-bcec-cbfbfeb60d30)
After some [investigation](https://www.codeproject.com/Articles/814885/MIDI-Monitor-Written-in-Visual-Basic), I've found info, that some mkb's yield midi-on on release,
>As a side note: I also found that my MIDI keyboard does not generate a Note Off message but a Note On with velocity '00' when releasing a key (see the screendump above).

Maybe this could be your case, so if you're still interested in finding out, could you check what events your mkb sends with any midi-monitor?
So I think that since the patch does work for your mkb and the original version doesn't, I'll merge it and create options file & section in app to allow users choose which comparison they want to use.

**Comment by fyrkantis** [comment]
I've tested my MIDI keyboard on [hardwaretester.com](https://hardwaretester.com/midi) the following results:

```
9721.00s: Chan 1 Note off C4 (128 60 64)
9337.00s: Chan 1 Note on C4 (144 60 64)
8776.90s: Chan 1 Note off C4 (128 60 64)
8498.00s: Chan 1 Note on C4 (144 60 72)
```

I hope this is includes the data you were asking about.

When I created the patch, I didn't understand the documentation but noticed `cmkp` for me was 64 whenever a key was released and something either higher or lower depending on the velocity when a key is pressed.

Would be a bit annoying if this is actually different for different MIDI keyboards, but if it is then creating and options file does sound like the best solution.
